lib/jekyll-gist/gist_tag.rb

```diff
@@ -62,20 +62,40 @@ def gist_noscript_tag(gist_id, filename = nil)
       end
 
       def fetch_raw_code(gist_id, filename = nil)
+        content = code_from_api(gist_id, filename)
+        return content if content
+
         url = "https://gist.githubusercontent.com/#{gist_id}/raw"
         url = "#{url}/#{filename}" unless filename.empty?
-        begin
-          uri = URI(url)
-          Net::HTTP.start(uri.host, uri.port,
-            use_ssl: uri.scheme == 'https',
-            read_timeout: 3, open_timeout: 3) do |http|
-            request = Net::HTTP::Get.new uri.to_s
-            response = http.request(request)
-            response.body
-          end
-        rescue SocketError, Net::HTTPError, Net::OpenTimeout, Net::ReadTimeout, TimeoutError
-          nil
+        uri = URI(url)
+        Net::HTTP.start(uri.host, uri.port,
+          use_ssl: uri.scheme == 'https',
+          read_timeout: 3, open_timeout: 3) do |http|
+          request = Net::HTTP::Get.new uri.to_s
+          response = http.request(request)
+          response.body
         end
+      rescue SocketError, Net::HTTPError, Net::OpenTimeout, Net::ReadTimeout, TimeoutError
+        nil
+      end
+
+      private
+
+      def code_from_api(gist_id, filename = nil)
+        return unless ENV["JEKYLL_GITHUB_TOKEN"]
```

Comment:
Good idea. I was thinking about other programs that use Octokit and what they use. Let's do `JEKYLL_GITHUB_TOKEN` for now – I think parity with `jekyll-github-metadata` for now makes sense. :) Thanks, Ben!